sbt_common/storage/src/main/scala/uk/ac/wellcome/storage/type_classes/StorageStrategy.scala

```diff
@@ -0,0 +1,56 @@
+package uk.ac.wellcome.storage.type_classes
+
+import java.io.{ByteArrayInputStream, InputStream}
+
+import io.circe.Json
+
+import scala.io.Source.fromInputStream
+import scala.util.hashing.MurmurHash3
+
+trait StorageStrategy[T] {
+  def get(t: T): (InputStream, String)
```

Comment:
Some comments explaining what this returns would be helpful, because it's super non-obvious.

We've discussed this enough, it might be worth that comment including _why_ we have this signature.